codegen/src/main/java/software/amazon/awssdk/codegen/poet/transform/protocols/JsonMarshallerSpec.java

```diff
@@ -102,6 +103,13 @@ protected FieldSpec operationInfoField() {
             initializationCodeBlockBuilder.add(".operationIdentifier($S)", shapeModel.getMarshaller().getTarget());
         }
 
+        String smithyProtocol = shapeModel.getMarshaller().getSmithyProtocol();
+        if (StringUtils.isNotBlank(smithyProtocol)) {
+            initializationCodeBlockBuilder.add(".putAdditionalMetadata($T.SMITHY_PROTOCOL, $S)",
+                                               OperationMetadataAttribute.class, smithyProtocol);
+        }
```

Comment:
Should we create codegen tests for CBOR services/marshallers?

Reply:
Yes, I will in a follow up PR. Let me know if you'd prefer to add them to this PR.

Reply:
If we have a mechanism to make sure we don't forget, that's fine. I usually prefer the tests in the PR, but I suppose you waited since this one was getting long.